js/Source/FlatUI.Select/FlatUI.Select.Filter.js

```diff
@@ -0,0 +1,90 @@
+/*
+---
+
+name: FlatUI.Select.Filter
+
+description: Filter input for FlatUI Select components.
+
+requires:
+ - FlatUI.Select
+ - Form.Filter
+
+
+provides: [FlatUI.Select.Filter]
+
+...
+*/
+
+
+FlatUI.Select.Filter = new Class({
+
+  Extends: FlatUI.Select,
+
+  build: function(){
+    this.parent.apply(this, arguments);
+    // build the DOM elements for our filter, inject them into the list UI
+    this.filterLi = new Element('li.filter').inject(this.list, 'top');
+    this.filterInput = new Element('input.form-control[placeholder=search]');
+    new Element('div.form-group.no-margin.padding-small').adopt(
+      this.filterInput
+    ).inject(this.filterLi);
+
+    // create an instance of our Form.Filter
+    this.formFilter = new Form.Filter(this.filterInput, {
+      items: '!ul li.flat-ui-select-li',
+      text: 'span'
+    });
+  },
+
+  destroy: function(){
+    this.parent.apply(this, arguments);
+    this.formFilter.detach();
+  },
+
+  reset: function(){
+    this.parent.apply(this, arguments);
+    // reset the filter, too
+    this.filterInput.set('value', '');
+    this.formFilter.filter();
+  },
+
+  show: function(e){
+    if (this.disabled || this.open) return;
+    this.parent.apply(this, arguments);
+    // on show, select the filter input, so the user can just start typing
+    // delay is required because of the transition
+    this.filterInput.select.delay(300, this.filterInput);
```

Comment:
I made the video before I added this line I think. Also, hotkeys make no sense on mobile, so I'm going to leave this here. Even if the transition is gone, it still functions well.